src/jit/codegenarmarch.cpp

```diff
@@ -1922,228 +1922,259 @@ void CodeGen::genCodeForCpBlkHelper(GenTreeBlk* cpBlkNode)
     }
 }
 
-// Generate code for InitBlk by performing a loop unroll
-// Preconditions:
-//   a) Both the size and fill byte value are integer constants.
-//   b) The size of the struct to initialize is smaller than INITBLK_UNROLL_LIMIT bytes.
-void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* initBlkNode)
+//----------------------------------------------------------------------------------
+// genCodeForInitBlkUnroll: Generate unrolled block initialization code.
+//
+// Arguments:
+//    node - the GT_STORE_BLK or GT_STORE_OBJ node to generate code for
+//
+void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
 {
+    assert(node->OperIs(GT_STORE_BLK, GT_STORE_OBJ));
+
 #ifndef _TARGET_ARM64_
     NYI_ARM("genCodeForInitBlkUnroll");
 #else
-    // Make sure we got the arguments of the initblk/initobj operation in the right registers
-    unsigned size    = initBlkNode->Size();
-    GenTree* dstAddr = initBlkNode->Addr();
-    GenTree* initVal = initBlkNode->Data();
-    if (initVal->OperIsInitVal())
-    {
-        initVal = initVal->gtGetOp1();
-    }
+    regNumber dstAddrBaseReg = genConsumeReg(node->Addr());
+    unsigned  dstOffset      = 0;
 
-    assert(dstAddr->isUsedFromReg());
-    assert((initVal->isUsedFromReg() && !initVal->IsIntegralConst(0)) || initVal->IsIntegralConst(0));
-    assert(size != 0);
-    assert(size <= INITBLK_UNROLL_LIMIT);
+    regNumber srcReg;
+    GenTree*  src = node->Data();
 
-    emitter* emit = GetEmitter();
+    if (src->OperIs(GT_INIT_VAL))
+    {
+        assert(src->isContained());
+        src = src->gtGetOp1();
+    }
 
-    genConsumeOperands(initBlkNode);
+    if (!src->isContained())
+    {
+        srcReg = genConsumeReg(src);
+    }
+    else
+    {
+        assert(src->IsIntegralConst(0));
```

Comment:
`src` is contained in this case so there's nothing to consume.

Reply:
`genConsumeRegs` (that was called here before through `genConsumeOperands` ) has 
```
        else if (tree->OperIsInitVal())
        {
            genConsumeReg(tree->gtGetOp1());
        }
```
that is probably dead after that change?

Reply:
> src is contained in this case so there's nothing to consume.

the diffs between `genConsumeReg` and `genConsumeRegs` are so confusing.

Reply:
> the diffs between genConsumeReg and genConsumeRegs are so confusing.

You can say that again :)

> genConsumeRegs (that was called here before through genConsumeOperands ) has

Ah, of course. It never made sense to be there to begin with since `GT_INIT_VAL` is such a special case.

Reply:
`GT_INIT_VAL` removed from `genConsumeOperands`